lib/SPIRVProducerPass.cpp

```diff
@@ -4434,6 +4461,158 @@ void SPIRVProducerPassImpl::GenerateInstruction(Instruction &I) {
         Ops.clear();
         Ops << Ty << Cmp << Mod << Modm1;
         RID = addSPIRVInst(spv::OpSelect, Ops);
+      } else if (I.getOpcode() == Instruction::FDiv && !Option::UnsafeMath()) {
+        // float div(float a, float b) {
```

Comment:
What happens with other types? Double for sure should be handled, but I'm not certain that half divide could appear.

Reply:
I agree, devices supporting those types we will need to handle them differently.
Let's reduce this implementation only for fp32 to start with.